Win/SketchfabUploader.h

```diff
@@ -0,0 +1,235 @@
+#include "stdafx.h"
+#include "Resource.h"
+#include "curl/include/picojson.h"
+#include "curl/include/curl.h"
+
+#define SKETCHFAB_SERVER "https://sketchfab.com"
+#define SKETCHFAB_MODELS SKETCHFAB_SERVER "/models"
+#define SKETCHFAB_MODELS_API SKETCHFAB_SERVER "/v2/models"
+
+typedef std::map<std::string, std::string> attributes;
+curl_off_t firstStep = -1;
+
+struct ProgressbarUpdater {
+  CURL *curl;
+  HWND progressBar;
+};
+
+// Taken from https://curl.haxx.se/libcurl/c/progressfunc.html
+static int progress_callback(void *clientp,
+  curl_off_t dltotal, curl_off_t dlnow,
+  curl_off_t ultotal, curl_off_t ulnow)
+{
+
+  UNREFERENCED_PARAMETER(dltotal);
+  UNREFERENCED_PARAMETER(dlnow);
+  // Hack to smooth the progress bar
+  if(firstStep == -1 && ulnow > 1000)
+     firstStep = ulnow;
+
+  struct ProgressbarUpdater *myp = (struct ProgressbarUpdater *)clientp;
+  double curtime = 0;
+
+  curl_easy_getinfo(myp->curl, CURLINFO_TOTAL_TIME, &curtime);
+
+  curl_off_t total_percen = 0;
+  if(ultotal > CURL_OFF_T_C(1)) {
+    total_percen = (ulnow - firstStep) / ((ultotal - firstStep)/ CURL_OFF_T_C(100));
+    total_percen = (total_percen - 50) * 2;
+  }
+
+  SendMessage(myp->progressBar, PBM_SETPOS, (WPARAM)float(total_percen), 0);
+
+  return 0;
+};
+
+static size_t WriteMemoryCallback(char *contents, size_t size, size_t nmemb, void *userp) {
+  ((std::string*)userp)->append((char*)contents, size * nmemb);
+  return size * nmemb;
+}
+
+class SketchfabV2Uploader {
+public:
+    SketchfabV2Uploader()
+    {}
+    ~SketchfabV2Uploader()
+    {}
```

Comment:
if you don't do anything, you're probably better off not defining the cstor/dstor and leave the compiler create those for you